### PR TITLE
Add licensing API

### PR DESCRIPTION
**Pull request by Roi-Hochler**
This branch should be published as a beta version for the licensing private preview

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -100,6 +100,34 @@ declare namespace powerbi {
...
+
+        /** Sentinel value. */
+        Unknown = 4
+    }
+
+    export const enum LicenseNotificationType {
+        /** Used by the visual to display an icon license notification, display "upgrade" button. */
+        General = 0,
+
+        /** Used by the visual to display an unsupported environment license notification */
```
Comment (jennyfel) [inline index.d.ts:125]: worth mention that "upgrade button" won't be displayed

**Review by jennyfel — APPROVED** [review]

File: index.d.ts [hunk 2]
```diff
@@ -100,6 +100,34 @@ declare namespace powerbi {
...
+        /** Indicates that the license is not active and shouldn't be used for provisioning benefits.  */
+        Inactive = 0,
+
+        /** Indicates that the license is active and can be used for provisioning benefits. */
+        Active = 1,
+
+        /** Indicates that the license is in grace period likely due to payment violation. */
+        Warning = 2,
+
+        /** Indicates that the license is suspended likely due to Payment violation. */
```
Comment (ramizsha) [inline index.d.ts:114]: payment (small p)

File: index.d.ts [hunk 3]
```diff
@@ -1601,6 +1639,22 @@ declare module powerbi.extensibility.visual {
...
+    }
+
+    export interface LicenseInfoResult {
+        /** An array of Service Plans purchased by the active user for this visual */
+        plans: ServicePlan[] | undefined;
+
+        /** Indicates that the visual is being rendered in a Power BI environment that doesn't support licenses management or enforcement. */
+        isLicenseUnsupportedEnv: boolean;
+
+        /** If false, indicates that the licenses info could not be retrieved. */
```
Comment (ramizsha) [inline index.d.ts:1655]: hmm this doesn't seem right - how about "Indicates whether the licenses info could be retrieved"

**Comment by Roi-Hochler** [comment]
This branch should be published as a beta version for the licensing private preview

**Review by AviSander — APPROVED** [review]

**Review by AviSander — APPROVED** [review]

**Pull request merged**